### PR TITLE
Added CUDA support for interpolation AA bilinear/bicubic

### PR DESCRIPTION
**Pull request by vfdev-5**
Description:

- Added CUDA support for interpolation AA bilinear/bicubic
- Updated tests

### CODE REVIEW

File: torchvision/csrc/ops/cuda/interpolate_aa_kernels.cu [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+      (rwidth >= 1.0) ? (interp_size * 0.5) * rwidth : interp_size * 0.5);
+
+    const int interp_height = (int)ceilf(support_h) * 2 + 1;
+    const int interp_width = (int)ceilf(support_w) * 2 + 1;
+
+    // Setup local buffers
+    scalar_t wx[256];
+    scalar_t wy[256];
+    scalar_t buffer1[256];
+    scalar_t buffer2[256];
```
Comment (vfdev-5) [inline torchvision/csrc/ops/cuda/interpolate_aa_kernels.cu:165]: Here is an assumption that resize scale is less than 127.

File: torchvision/csrc/ops/cuda/interpolate_aa_kernels.cu [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+        const accscalar_t rheight = area_pixel_compute_scale<accscalar_t>(
+            input_height, output_height, align_corners, scales_h);
+        const accscalar_t rwidth = area_pixel_compute_scale<accscalar_t>(
+            input_width, output_width, align_corners, scales_w);
+
+        // We are using static buffer memory of 256 * sizeof(float) per thread
+        // to store weights. Size of weights array is
+        // interp_size = scale * 2 + 1 for bilinear mode
+        TORCH_CHECK(rheight < 127, "Max supported scale factor is 127");
+        TORCH_CHECK(rwidth < 127, "Max supported scale factor is 127");
```
Comment (fmassa) [inline torchvision/csrc/ops/cuda/interpolate_aa_kernels.cu:272]: Is the max scale factor supported different if we use `bicubic`?

**Review by fmassa — COMMENTED** [review]
Thanks for the quick PR Victor!

The PR looks good to merge, I just have one question before, can you have a look and let me know what you think?

File: torchvision/csrc/ops/cuda/interpolate_aa_kernels.cu (hunk 2 above)
Comment (vfdev-5) [reply torchvision/csrc/ops/cuda/interpolate_aa_kernels.cu:272]: Thanks for catching that, yes for bicubic max scale should be at most 63. In general the check should be
```c++
TORCH_CHECK(rheight < 255 / interp_size, "Max supported scale factor is 127 (bilinear), 64 (bicubic)");
```

**Review by fmassa — APPROVED** [review]
LGTM, thanks!

Can you rebase the PR so that we can proceed merging it?

**Pull request merged**